core/src/main/java/com/dtolabs/rundeck/core/execution/ExecutionServiceImpl.java

```diff
@@ -414,10 +414,17 @@ public NodeExecutorResult executeCommand(final ExecutionContext context, final E
                 .nodeContextData(node)
                 .build();
 
+        //It tries to get the interpreter from the node attributes first, then from the project properties.
+        String commandInterpreter = node.getAttributes().get("winrm-shell") != null ?
+                node.getAttributes().get("winrm-shell")
+                : context.getIFramework().getFrameworkProjectMgr()
+                        .getFrameworkProject(context.getFrameworkProject()).getProperty("project.plugin.NodeExecutor.WinRMPython.shell");
+
```

Comment:
Should we have this at this point? This is related to each resource mode + node-executor plugin. Why if we use a different Windows plugin, such as the Powershell plugin?